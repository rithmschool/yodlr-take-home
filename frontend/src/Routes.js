import React from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import Home from './Home';
import UsersTable from './UsersTable';
import Signup from './Signup';

const Routes = () => (
  <Switch>
    <Route exact path="/admin">
      <UsersTable />
    </Route>
    <Route exact path="/signup">
      <Signup />
    </Route>
    <Route exact path="/">
      <Home />
    </Route>
    <Redirect to="/" />
  </Switch>
)


export default Routes;