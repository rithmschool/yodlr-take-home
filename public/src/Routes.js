import React from 'react';
import { Route, BrowserRouter, Switch } from 'react-router-dom';
import PrivateRoute from './PrivateRoute';
import Home from './Home';
import Admin from './Admin';
import NavBar from './NavBar';
import User from './User';
import Login from './Login';
import Signup from './Signup';

function Routes ({ logout, login, signup }) {

    return (
        <div className="Routes">
          <BrowserRouter>
            <NavBar logout={logout}/>
            <main>
              <Switch>
                <Route exact path="/">
                  <Home />
                </Route>
                
                <PrivateRoute exact path="/users/:id" component={User}/>
                <Route exact path="/login">
                  <Login login={login}/>
                </Route>
                <Route exact path="/admin"><Admin/></Route>
                <PrivateRoute exact path="/admin" component={Admin}/>
                <Route exact path="/signup">
                  <Signup signup={signup}/>
                </Route>
                <Route>
                  <p>Hmmm. I can't seem to find what you want.</p>
                </Route>
              </Switch>
            </main>
          </BrowserRouter>
        </div>
      );
}

export default Routes;