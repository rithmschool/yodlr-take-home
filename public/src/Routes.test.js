import React from "react";
import { render } from "@testing-library/react";
import Routes from './Routes';

it("renders without crashing", function() {
  render(<Routes />);
});

it("matches snapshot", function() {
  const {asFragment} = render(<Routes />);
  expect(asFragment()).toMatchSnapshot();
});