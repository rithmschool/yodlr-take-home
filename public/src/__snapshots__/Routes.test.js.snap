// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Routes"
  >
    <div>
      <nav
        class="navbar navbar-expand-md navbar-light bg-light"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Yodlr
        </a>
        <ul
          class="mr-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/signup"
            >
              Register
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/login"
            >
              Login
            </a>
          </li>
          <div />
        </ul>
      </nav>
    </div>
    <main>
      <div>
        <h3>
          Welcome to the Yodlr Design Challenge!
        </h3>
        <p>
          Click 'Sign Up' to register.
        </p>
        <a
          class="btn btn-primary"
          href="/signup"
        >
          Sign Up
        </a>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
